src/cm_stats_api.h

```diff
@@ -70,6 +70,15 @@ stats_ns_t *
 stats_ns_t *
   stats_register_ns(stats_recorder_t *, stats_ns_t *, const char *);
 
+/* Deregister and free a namespace.
+ * Results are undefined if ns is still being used during or after this call
+ * if the return value is true.
+ * This function deregisters ns from parent within rec. If successful, ns is
+ * freed.
+ */
```

Comment:
What do you think about separating the `deregister` from the `free`? I'm thinking we could deregister a namespace as part of a tear-down process, then fold the statistics under that namespace into a namespace dedicated to historical data (accumulating all the stuff that had been torn down in the past), and then free. It also provides a mechanism to move metrics from one location to another, like if we wanted to have "in-progress" metrics retrievable, and then committed to a new place?